### PR TITLE
feat: derive Clone, PartialEq and Eq and add into_subparts

### PR DESCRIPTION
**Pull request by dignifiedquire**
This solves all issues I reported in #47

### CODE REVIEW

**Comment by staktrace** [comment]
So one potential problem I see here is that the meaning of `Eq` and `PartialEq` is not defined in writing. Consider:

```
let (header1, _) = parse_header(b"Key: Value").unwrap();
let (header2, _) = parse_header(b"Key: =?iso-8859-1?Q?Value?=").unwrap();
```

Here using `==` to compare `header1` and `header5` will return false even though the `get_key()` and `get_value()` functions return identical strings. In other words even though the headers are semantically equal the `Eq` implementation doesn't reflect that. This may be confusing to users of this code and result in unexpected behavior unless it is clearly documented. I assume that for your use case you don't care about this distinction but in the general case it may matter.

File: src/lib.rs [hunk 1]
```diff
@@ -749,6 +749,12 @@ impl<'a> ParsedMail<'a> {
             .unwrap_or_default();
         Ok(disposition)
     }
+
+    /// Returns the subparts, consuming the existing structure, allowing them to be
+    /// processed further independently.
+    pub fn into_subparts(self) -> Vec<ParsedMail<'a>> {
+        self.subparts
```
Comment (staktrace) [inline src/lib.rs]: You can just do this from the call site since the subparts field is public. What does this function buy you?

Comment (dignifiedquire) [reply src/lib.rs]: You are right, I don't know why I didn't realize this earlier.

**Comment by dignifiedquire** [comment]
> So one potential problem I see here is that the meaning of `Eq` and `PartialEq` is not defined in writing.

Right, fair enough. I am interested in knowing if the underyling raw message was the same at the end of the day, but I can see how this behaviour could be confusing

**Comment by staktrace** [comment]
If you still want the `Eq` as you implemented, I'd be ok with merging that change if the behaviour is documented on the various structs.

**Comment by staktrace** [comment]
I'm going to close this for now. We can reopen later if needed.

**Comment by dignifiedquire** [comment]
Sounds good, will write if I end up neeeding to push this forward. So far I could make things work without it

**Pull request closed without merging**